Specific social shares take 2

hsribei opened the pull request:
Taking social meta tag information from Youtube nodes.

This assumes that the lwyp format is always video+article, that is, that every lesson's `.md` has a valid `videoId` frontmatter field. (This is true for all current ones.)

**Important:** at the moment, `gatsby-node.js` only creates `ytVideo` nodes for the `react` playlist, not the `state` playlist. I'm going to fix that in a separate PR, making it easy to add new playlists. That PR should be merged before this one.

hsribei committed:
Copy share buttons from landing page

hsribei committed:
FacebookShareButton doesn't accept title pre-fill

hsribei committed:
Add social share dialog with article-specific text

hsribei committed:
Pass url to ShareDialog as prop

hsribei committed:
Use yt title, thumb & desc in social meta tags

hsribei committed:
Add `npm run debug-develop`

hsribei committed:
Move ShareDialog into a component file

hsribei committed:
Fix weird textarea extra height bug

Swizec commented:
This looks great!

Altho I think I've found a bug. On the homepage at least, the `twitter:image` tag has the wrong URL.

```
<meta name="twitter:image" content="https://learnwhileyoupoop.comhttp://localhost:8000/static/social-f14cb240d58495f905d9e7ef85fbe8a2.png" data-react-helmet="true">
```

Same for `og:image`

```
<meta property="og:image" content="https://learnwhileyoupoop.comhttp://localhost:8000/static/social-f14cb240d58495f905d9e7ef85fbe8a2.png" data-react-helmet="true">
```

Swizec commented:
The assumption that every lesson has a youtube link and an MD is correct.

Swizec commented:
Also for some reason I can't load specific articles while on this branch. Is that because I merged the other branch you mention, but this one doesn't include it?

I tried on `/state/*` lessons

hsribei commented:
> I tried on `/state/*` lessons

Yes, you need #8 in order to have the graphql nodes for the `state` youtube videos. Since you already merged it, I rebased on it and now you should be able to see `state` articles when checking out this branch.

Btw, master is broken for me right now, so I rebased it on master~1, which is the #8 merge commit.

hsribei commented:
>Altho I think I've found a bug. On the homepage at least, the twitter:image tag has the wrong URL.

You're right about this, though I'm seeing this on master as well without this branch. I'll track it down.

hsribei commented:
>You're right about this, though I'm seeing this on master as well without this branch. I'll track it down.

Fixed in #9
